### PR TITLE
feat: command-line overstrike detection with mode()/nvim_get_mode()

### PR DESCRIPTION
**Pull request by Sam-programs**
i found a way to redraw with cmd preview is on without flickering. but it needs to be in replace mode to work.
`<left>(character before the cursor)`
overstriking the character before the cursor to position the cursor causing a refresh and not changing the actual text in the command line.

detecting overstrike by changing the command line causes flickering.
i also think adding a way to detect this will allow command line mappings to be more consistent.

### CODE REVIEW

**Comment by clason** [comment]
Why did you delete `.clang-tidy`?

**Comment by Sam-programs** [comment]
`git -A`
removed it

**Comment by clason** [comment]
Well, then don't do that ;)

**Comment by clason** [comment]
Also, maybe this would be good to PR to Vim first?

**Comment by Sam-programs** [comment]
maybe ill make a pr there ig

**Comment by Sam-programs** [comment]
How do i fix this?
> FAIL: /home/runner/work/neovim/neovim/src/nvim/state.c (File size changed from 9192 to 9193)

File: src/nvim/state.c [hunk 1]
```diff
@@ -211,6 +212,9 @@ void get_mode(char *buf)
     if (exmode_active) {
       buf[i++] = 'v';
     }
+    if (cmdline_overstrike()){
```
Comment (zeertzjq) [inline src/nvim/state.c:215]:
```suggestion
    if (cmdline_overstrike()) {
```

**Comment by clason** [comment]
`make formatc`

**Comment by Sam-programs** [comment]
Sorry.

**Comment by clason** [comment]
...aaaand it's gone again

**Comment by Sam-programs** [comment]
It is ?

**Comment by clason** [comment]
yes.

**Comment by Sam-programs** [comment]
mb.

**Comment by Sam-programs** [comment]
um was i supposed to define the modes for the screen tests?

**Comment by Sam-programs** [comment]
rerunning the tests removes 2 failing tests and a new unrelated failing test.
> Error: The action has timed out.
why?
is the 'i' insert?
```
    sendkeys('i:e tmp_terminal_nvim<Enter>')
```
and that is slowing it down?
but it didn't fail last time.

**Comment by Sam-programs** [comment]
YES.

**Comment by clason** [comment]
https://github.com/vim/vim/pull/13569

Closing this PR since we'll port the Vim patch (including the tests) when it is merged. Thank you!

**Pull request closed without merging**